tools/image_processing/imagej2/imagej2_bunwarpj_compare_elastic/imagej2_bunwarpj_compare_elastic.py

```diff
@@ -0,0 +1,72 @@
+parser.add_argument( '--target_image_format', dest='target_image_format', help='Target image format' )
+parser.add_argument( '--source_transformation', dest='source_transformation', default=None, help='Direct source transformation matrix' )
+parser.add_argument( '--target_transformation', dest='target_transformation', default=None, help='Inverse target transformation matrix' )
+parser.add_argument( '--max_heap_size_type', dest='max_heap_size_type', help='Type (default or megabytes) of max_heap_size value' )
+parser.add_argument( '--max_heap_size', dest='max_heap_size', help='Maximum size of the memory allocation pool used by the JVM.' )
+parser.add_argument( '--output', dest='output', help='Warping index' )
+
+args = parser.parse_args()
+
+tmp_dir = imagej2_base_utils.get_temp_dir()
+source_image_path = imagej2_base_utils.get_input_image_path( tmp_dir, args.source_image, args.source_image_format )
+target_image_path = imagej2_base_utils.get_input_image_path( tmp_dir, args.target_image, args.target_image_format )
+source_transformation_path = imagej2_base_utils.get_input_image_path( tmp_dir, args.source_transformation, 'txt' )
+target_transformation_path = imagej2_base_utils.get_input_image_path( tmp_dir, args.target_transformation, 'txt' )
+# bUnwarpJ produces several lines of output that we need to discard, so
+# we'll use a temporary output file from which we'll read only the last line.
+tmp_output_path = imagej2_base_utils.get_input_image_path( tmp_dir, args.output, 'txt' )
+
+# Set the size of the memory allocation pool used by the JVM.
+memory_size = imagej2_base_utils.get_max_heap_size_value( args.max_heap_size_type, args.max_heap_size )
+
+# Define command response buffers.
+tmp_out = tempfile.NamedTemporaryFile().name
+tmp_stdout = open( tmp_out, 'wb' )
+tmp_err = tempfile.NamedTemporaryFile().name
+tmp_stderr = open( tmp_err, 'wb' )
+
+# Build the command line to align the two images.
+cmd = imagej2_base_utils.get_base_cmd_bunwarpj( memory_size )
+if cmd is None:
+    imagej2_base_utils.stop_err( "bUnwarpJ not found!" )
+cmd += ' -compare_elastic'
+# Target is sent before source.
+cmd += ' %s' % target_image_path
+cmd += ' %s' % source_image_path
+cmd += ' %s' % target_transformation_path
+cmd += ' %s' % source_transformation_path
+cmd += ' > %s' % tmp_output_path
+
+# Calculate the warping index of two elastic transformations using bUnwarpJ.
+proc = subprocess.Popen( args=cmd, stderr=subprocess.PIPE, stdout=subprocess.PIPE, shell=True )
+rc = proc.wait()
+if rc != 0:
+    error_message = imagej2_base_utils.get_stderr_exception( tmp_err, tmp_stderr, tmp_stdout )
+    imagej2_base_utils.stop_err( error_message )
+
+# Example contents of tmp_output_path:
+# Target image : ~/tmpYOkRxE.jpg
+# Source image : ~/tmpVVtDiE.gif
+# Target Transformation file : ~/tmpi5338f.txt
+# Source Transformation file : ~/tmplVWe3l.txt
+# Warping index = 14.87777347388348
+results = open( tmp_output_path, 'r' ).readlines()
```

Comment:
@bgruening The file will always be about the size of the example contents I listed - just 5 lines, although the paths will be longer, so I felt deadlines() would be safe.  I will definitely change this to iterate if you'd like me to.  Please confirm.  ;)